src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java

```diff
@@ -230,13 +285,179 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository, Parti
+                                ((MultipleChoiceQuestion) originalQuestion).getAnswerOptions().size()) {
+                                updateResultsAndStatistics = true;
+                            }
+                        }
+                    }
+                }
+            } else {
+                // question is added (not allowed)
+                addedQuestions.add(question);
+            }
+        }
+        // remove all added questions
+        quizExercise.getQuestions().removeAll(addedQuestions);
+
+        //update QuizExercise
+        ResponseEntity<QuizExercise> methodResult = updateQuizExercise(quizExercise);
+
+        // question deleted?
+        if (quizExercise.getQuestions().size() != originalQuizExercise.getQuestions().size()) {
+            updateResultsAndStatistics = true;
+        }
+
+        // update Statistics and Results
+        if (updateResultsAndStatistics) {
+
+            //reset all statistic
+            quizExercise.getQuizPointStatistic().resetStatistic();
+            for (Question question : quizExercise.getQuestions()) {
+                question.getQuestionStatistic().resetStatistic();
+            }
+
+            for (Participation participation : participationRepository.findByExerciseId(quizExercise.getId())) {
+
+                Result lastRatedResult = null;
+                Result lastUnratedResult = null;
+
+                for (Result result : resultRepository.findByParticipationIdOrderByCompletionDateDesc(participation.getId())) {
+
+                    //recalculate existing score
+                    ((QuizSubmission) result.getSubmission()).calculateAndUpdateScores(quizExercise);
+                    result.setScore(Math.round(((QuizSubmission) result.getSubmission()).getScoreInPoints() / quizExercise.getMaxTotalScore() * 100));
+                    if (result.getScore() == 100) {
+                        result.setSuccessful(true);
+                    } else {
+                        result.setSuccessful(false);
+                    }
+                    resultRepository.save(result);
+                    quizSubmissionRepository.save((QuizSubmission) result.getSubmission());
+
+                    //save last rated and unrated Result for Statistic update
+                    if (result.getCompletionDate().isBefore(quizExercise.getDueDate().plusSeconds(5))
+                        && (lastRatedResult == null || lastRatedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
+                        lastRatedResult = result;
+                    }
+                    if (result.getCompletionDate().isAfter(quizExercise.getDueDate().plusSeconds(5))
+                        && (lastUnratedResult == null || lastUnratedResult.getCompletionDate().isBefore(result.getCompletionDate()))) {
+                        lastUnratedResult = result;
+                    }
+                }
+
```

Comment:
again the whole code is too complex and unstructured. Please try to extract method with a special purpose and definitely add more comments